.github/workflows/go-releaser.yml

```diff
@@ -12,28 +12,248 @@ permissions:
+        path: linux-package-feeds-repo
+
+    - run: |
+        ln -s linux-package-feeds-repo/source linux-package-feeds
+
+    - uses: actions/download-artifact@v3
+      with:
+        name: artifacts
+        path: artifacts/
+
+    - name: Download nuget.exe
+      run: curl -o ./nuget.exe https://dist.nuget.org/win-x86-commandline/latest/nuget.exe
+
+    - name: Create Chocolatey package
+      env:
+        MSI_FILE: ${{ needs.msi.outputs.msi_file }}
+      run: |
+        cp artifacts/$(basename "${MSI_FILE//\\//}") build/tools
+        mono nuget.exe pack ./build/cli.nuspec -Version ${{ needs.goreleaser.outputs.version }} -OutputDirectory artifacts/
+
+    - name: Compress packages
+      run: |
+
+        zip -rj artifacts/octopus-cli.${{ needs.goreleaser.outputs.version }}.zip \
+          artifacts/*.nupkg \
+          artifacts/*macOS*amd64.tar.gz \
+          artifacts/*.deb \
+          artifacts/*.rpm \
+          linux-package-feeds/publish-*.sh
+
+    - uses: actions/upload-artifact@v3
+      with:
+        name: octopus-cli.${{ needs.goreleaser.outputs.version }}
+        path: artifacts/octopus-cli.${{ needs.goreleaser.outputs.version }}.zip
+
+  publish:
+    needs: [goreleaser, generate-packages]
+    runs-on: ubuntu-latest
+    env:
+      OCTOPUS_API_KEY: ${{ secrets.OCTOPUS_API_KEY }}
+      OCTOPUS_HOST: ${{ secrets.OCTOPUS_URL }}      
+      OCTOPUS_CLI_SERVER: ${{ secrets.OCTOPUS_URL }}
+      OCTOPUS_CLI_API_KEY: ${{ secrets.OCTOPUS_API_KEY }}
+
+    steps:
+    - uses: actions/download-artifact@v3
+      with:
+        name: octopus-cli.${{ needs.goreleaser.outputs.version }}
+
+    - name: Install Octopus CLI 🐙
+      uses: OctopusDeploy/install-octopus-cli-action@v1
+      with:
+        version: latest
+
+    - name: Push a package to Octopus Deploy 🐙
+      uses: OctopusDeploy/push-package-action@v2
+      with:
+        space: Integrations
+        packages: octopus-cli.${{ needs.goreleaser.outputs.version }}.zip
+
+    - name: Fetch Release Notes
+      id: fetch-release-notes
+      if: "!contains(needs.goreleaser.outputs.version, '-')"
+      run: |
+        echo "::debug::${{github.event_name}}"
+        OUTPUT_FILE="release_notes.txt"
+        gh view release "${{ needs.goreleaser.outputs.tag_name }}" --jq '.body' --json 'body' | sed 's#\r#  #g' > $OUTPUT_FILE
+        echo "::set-output name=release-note-file::$OUTPUT_FILE"
+
+    - name: Create a release in Octopus Deploy 🐙
+      uses: OctopusDeploy/create-release-action@v2
+      if: "!contains(needs.goreleaser.outputs.version, '-')"
+      with:
+        space: Integrations
+        project: 'cli'
+        package_version: ${{ needs.goreleaser.outputs.version }}
+        packages: '*:NuGet.CommandLine:4.4.1'
+        release_notes_file: ${{ steps.fetch-release-notes.outputs.release-note-file || ''}}
+        git_ref: ${{ (github.ref_type == 'tag' && github.event.repository.default_branch ) || (github.head_ref || github.ref) }}
```

Comment:
Am I reading this right? It looks like

git ref = 
   if github.ref_type is 'tag' && the repo has a default branch 
      then use the repo default branch
   else if github.head_ref is set 
     then use github.head_ref
   else 
      use github.ref
      
Why don't we just use github.ref all the time?

Reply:
@slewis74 do you remember the reason why we had to do this logic, I can't remember the reason but know we use it in most (all) our other builds as well...

Reply:
So this is why:
- For tags, `github.ref` is the tag ref i.e. `refs/tags/<tag_name>` so we'd have to use the default branch.
- For pull requests, `github.ref` is the merge branch for the pull request i.e. `refs/pull/<pr_number>/merge` so we have to use `github.head_ref` as it's the branch of the PR

In this workflow, we can simplify this by just using `github.default_branch` as the workflow is only fired when a tag is pushed.